### PR TITLE
[BACKLOG-9599] Several VizAPI accessibility changes

### PR DESCRIPTION
**Pull request by dcleao**
* Added Type#sourceId to complement Type#id in cases where the physical module layout
  does not match the type id.
* Changed Type#view to only consider relative ids starting with a "." (and not ending with ".js")
  and to make paths relative to the parent module of Type#id.
* Changed the ids of type and CCC wrapper classes to use module.id
* Changed the CCC wrapper classes to specify Type#sourceId.
* Renamed Type#view and Type#viewClass to Type#defaultView and Type#defaultViewClass
* Added pentaho.visual.base.View.createAsync helper method.
* Default Type#styleClass from Type#id.
  * Added missing Refinement.Type#inheritedStyleClasses implementation
  * Removed explicit styleClass values from type classes, CCC wrapper and samples
* Default the debug name of an Instance type from Type#id
  * Removed debug names from type classes

Needs review of JsDocs doclets: [this commit](https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/811/commits/66978e49f30454dfaad800d3b6d180020f144436).

@pentaho/millenniumfalcon please review.

### CODE REVIEW

**Comment by dcleao** [comment]
1. Container application code should be changed to use the new `View.createAsync` method.
2. Visualization models using a "viz in a folder" layout should use the new attribute `Model.Type#sourceId`, beside the `Model.Type#id`, whenever the actual model file module id is different from the logical id specified in `Model.Type#id` (when AMD package or explicit AMD mapping is used).
3. Visualization models should rename the attribute `Model.Type#view` to `Model.Type#defaultView` and change its value, when relative, to be relative to the parent module folder.

File: package-res/resources/web/pentaho/type/_doc/spec/ITypeProto.jsdoc [hunk 1]
```diff
@@ -150,6 +150,13 @@
  *
  * This attribute is typically used to associate an icon with a type.
  *
+ * When `null` or an empty string,
+ * the attribute value is set to `null`.
+ *
+ * When unspecified, or `undefined`,
+ * the attribute value defaults to the type's `id`,
+ * in _snake-case_, if it has one, or `null`, if not.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/type/_doc/spec/ITypeProto.jsdoc]: in _snake-case_, if it has one; or, `null`, if not.

File: package-res/resources/web/pentaho/type/_type.js [hunk 2]
```diff
@@ -414,6 +418,27 @@ define([
...
       _sourceId: null,
 
+      /**
+       * Gets the source module identifier of this type.
+       *
+       * The source identifier is the module id of the _actual_ AMD/RequireJS module
+       * that provides the type and may be different from the [identifier]{@link pentaho.type.Type#id}
+       * when an AMD package or custom mapping is configured for the module.
+       *
+       * The source identifier is used to resolve module ids relative to the source module,
```
Comment (jheck123) [inline package-res/resources/web/pentaho/type/_type.js]: The source identifier is used to resolve module identifiers relative to the source module,

File: package-res/resources/web/pentaho/type/_type.js [hunk 3]
```diff
@@ -461,6 +486,21 @@ define([
...
+      /**
+       * Builds an absolute module identifier from
+       * one that is relative to the type's [source location]{@link pentaho.type.Type#sourceId}.
+       *
+       * Relative module identifiers start with a `.` and do not end with `".js"`.
+       * For example, `"./View"` and `"../View"`, but not `./View.js`.
+       *
+       * Absolute identifiers are returned without modification.
+       *
+       * @param {string} id A module identifier.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/type/_type.js]: @param {string} id - A module identifier.

File: package-res/resources/web/pentaho/type/_type.js [hunk 4]
```diff
@@ -774,7 +814,13 @@ define([
        * Attempting to set to a non-string value type implicitly
        * converts the value to a string before assignment.
        *
-       * An empty string or `undefined` value is interpreted as `null`.
+       * An empty string or `null` clears the property value.
+       *
+       * Setting to `undefined`, makes the property assume its default value.
+       *
+       * The default value of a type with an [id]{@link pentaho.type.Type#id} is
+       * the id converted to _snake-case_.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/type/_type.js]: the identifier converted to _snake-case_.

File: package-res/resources/web/pentaho/type/_type.js [hunk 5]
```diff
@@ -927,12 +971,12 @@ define([
       /**
        * Gets a promise for the default view class or factory, if any, or `null`.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/type/_type.js]: Gets a promise for the default view class or factory, if any; or `null`.

File: package-res/resources/web/pentaho/util/text.js [hunk 6]
```diff
@@ -57,6 +57,15 @@ define(function() {
     },
 
     /**
+     * Convert a string into snake-like string.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/util/text.js]: Convert a string into a snake-like string.

File: package-res/resources/web/pentaho/util/text.js [hunk 7]
```diff
@@ -57,6 +57,15 @@ define(function() {
...
+     *
+     * Slashes and underscores are converted to an hyphen character, `-`.
+     * Camel-case strings are split and joined by an hyphen character as well.
+     *
+     * @param {string} name The string to convert to snake-case.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/util/text.js]: @param {string} name - The string to convert to snake-case.

File: package-res/resources/web/pentaho/visual/base/View.js [hunk 8]
```diff
@@ -368,6 +368,22 @@ define([
...
 
+    /**
+     * Creates a view for the given model, of the registered default type, if any.
+     *
+     * If no default view type is registered
+     * for the given model type,
+     * or for any of its base model types,
+     * the returned promise is rejected.
+     *
+     * @param {!pentaho.visual.base.Model} model The visual model.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/visual/base/View.js]: @param {!pentaho.visual.base.Model} model - The visual model.

File: package-res/resources/web/pentaho/visual/base/View.js [hunk 9]
```diff
@@ -368,6 +368,22 @@ define([
...
+     * @param {...any} other Other arguments to pass to the registered view constructor.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/visual/base/View.js]: @param {...any} other - Other arguments to pass to the registered view constructor.

File: package-res/resources/web/pentaho/visual/base/View.js [hunk 10]
```diff
@@ -368,6 +368,22 @@ define([
...
+     *
+     * @return {!Promise.<pentaho.visual.base.View>} A promise for a view of the given model.
+     *
+     * @rejects {pentaho.lang.ArgumentRequiredError} When `model` is not specified.
+     * @rejects {Error} When there is not registered default view class for the type of `model`.
```
Comment (jheck123) [inline package-res/resources/web/pentaho/visual/base/View.js]: @rejects {Error} When there is not a registered default view class for the type of `model`.

File: package-res/resources/web/pentaho/type/_type.js [hunk 11]
```diff
@@ -414,6 +418,27 @@ define([
 
       _sourceId: null,
 
+      /**
+       * Gets the source module identifier of this type.
+       *
+       * The source identifier is the module id of the _actual_ AMD/RequireJS module
```
Comment (jheck123) [inline package-res/resources/web/pentaho/type/_type.js]: The source identifier is the module identifier of the _actual_ AMD/RequireJS module

**Comment by dcleao** [comment]
Need to merge this now, cause others are dependent on having this by Monday.
Despite this, please make any comments and I will make any needed changes in a subsequent PR.
Thanks.

**Pull request merged**